system-contracts/bootloader/tests/bootloader/bootloader_test.yul

```diff
@@ -1,65 +1,154 @@
+     testing_assertEq(safeSub(10, 8, "err"), 2, "Failed to subtract 8")
+ }
+
+ function TEST_safeDiv() {
+     testing_assertEq(safeDiv(4, 2, "err"), 2, "Simple division")
+     testing_assertEq(safeDiv(5, 2, "err"), 2, "Rounding")
+     testing_assertEq(safeDiv(5, 3, "err"), 1, "Rounding down")
+     testing_assertEq(safeDiv(4, 3, "err"), 1, "Rounding down")
+     testing_assertEq(safeDiv(0, 3, "err"), 0, "Rounding down")
+ }
+
+ function TEST_safeDivAssert() {
+     testing_testWillFailWith("divByZero")
+     safeDiv(4, 0, "divByZero")
+ }
+
+ function TEST_asserts() {
+     testing_testWillFailWith("willFail")
+     safeSub(10, 12, "willFail")
+ }
+
+ function TEST_safeMul() {
+     testing_assertEq(safeMul(4, 2, "err"), 8, "Simple")
+     testing_assertEq(safeMul(0, 2, "err"), 0, "With zero")
+     testing_assertEq(safeMul(0, 0, "err"), 0, "With zero")
+     testing_assertEq(safeMul(2, 0, "err"), 0, "With zero")
+ }
+
+ function TEST_safeMulAssert() {
+     testing_testWillFailWith("overflow")
+     let left := shl(129, 1)
+     testing_log("left", left)
+     safeMul(left, left, "overflow")
+ }
+
+ // function TEST_should ignore
+
+ function TEST_strLen() {
+     testing_assertEq(getStrLen("abcd"), 4, "short string")
+     testing_assertEq(getStrLen("00"), 2, "0 filled string")
+     testing_assertEq(getStrLen(""), 0, "empty string")
+     testing_assertEq(getStrLen("12345678901234567890123456789012"), 32, "max length")
+     testing_assertEq(getStrLen("1234567890123456789012345678901234"), 0, "over max length")
+ }
+
+ function TEST_simple_transaction() {
+     // We'll test the transaction from 0.json
+     let txDataOffset := testing_txDataOffset(0)
+     let innerTxDataOffset := add(txDataOffset, 0x20)
+     testing_assertEq(getGasPerPubdataByteLimit(innerTxDataOffset), 0xC350, "Invalid pubdata limit")
+ }
+
+ function TEST_getTransactionUpfrontOverhead() {
+     // For very large transactions it should be proportional to the memory,
+     // but for small ones, the transaction slots are more important
+
+     let smallTxOverhead := getTransactionUpfrontOverhead(32)
+     let largeTxOverhead := getTransactionUpfrontOverhead(1000000)
+
+     testing_assertEq(smallTxOverhead, TX_SLOT_OVERHEAD_GAS(), "Invalid small tx overhead")
+     testing_assertEq(largeTxOverhead, mul(1000000, MEMORY_OVERHEAD_GAS()), "Invalid small tx overhead")
+ }
+
+ function TEST_ceilDiv() {
+     testing_assertEq(ceilDiv(1, 0), 0, "Dividing by 0")
+     testing_assertEq(ceilDiv(0, 1), 0, "Dividing with overflow")
+     testing_assertEq(ceilDiv(5, 2), 3, "Invalid division")
+     testing_assertEq(ceilDiv(6, 2), 3, "Invalid division")
+ }
+
+ function TEST_lengthRoundedByWords() {
+     testing_assertEq(lengthRoundedByWords(65), 96, "Invalid word length")
+ }
+
+ function TEST_getGasPrice() {
+     let baseFee := basefee()
+     testing_log("Base Fee", baseFee)
+     testing_assertEq(getGasPrice(6, 5), baseFee, "Invalid gas price")
```

Comment:
it does not test any of the possible errors in this function. I don't say that it is easy to add (not sure), but this can not be counted as test coverage. If adding is too hard, we can merge as is, it is better than nothing

Reply:
Please see my [other comment](https://github.com/matter-labs/era-contracts/pull/1300#discussion_r1984589046
).